flake.nix

```diff
@@ -80,7 +80,12 @@
               ];
             };
 
-            nixos = base: custom base.config.system.build.toplevel "$PROFILE/bin/switch-to-configuration switch";
+            nixos = base: custom base.config.system.build.toplevel ''
+              $PROFILE/bin/switch-to-configuration switch
+
+              # https://github.com/serokell/deploy-rs/issues/31
+              sed -i '/^default /d' ${base.config.boot.loader.efi.efiSysMountPoint}/loader/loader.conf || :
```

Comment:
I'm not sure of that actually, I haven't tested it but I know that GRUB boots the newest `deploy-rs`-deployed profile at least _sometimes_ (when no old generations have been garbage collected), so it would be weird if it acts like systemd-boot only sometimes

Reply:
Ah, ok, you're right. GRUB does the right thing: https://github.com/NixOS/nixpkgs/blob/master/nixos/modules/system/boot/loader/grub/install-grub.pl#L491

It doesn't check whether `/nix/var/nix/profiles` contains `$defaultConfig`